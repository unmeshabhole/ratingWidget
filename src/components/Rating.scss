.flex {
  display: flex;
}

.showDiv {
  display: contents;
}

.hideDiv {
  display: none;
}

.color-orange {
  color: orange;
}

.color-grey {
  color: grey;
}
