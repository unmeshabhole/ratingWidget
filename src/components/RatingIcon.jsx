import React from "react";
import "./RatingIcon.scss";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faStar } from "@fortawesome/free-solid-svg-icons";

function RatingIcon(props) {
  const { index, rating, onSaveRating } = props;
  const fill = rating >= index;
  return (
    <div className="cursor-pointer" onClick={() => onSaveRating(index)}>
      <FontAwesomeIcon
        icon={faStar}
        className={fill ? "color-orange" : "color-grey"}
      />
    </div>
  );
}
export default RatingIcon;
