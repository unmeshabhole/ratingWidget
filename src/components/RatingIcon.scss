.color-orange {
  color: orange;
}

.color-grey {
  color: grey;
}

.cursor-pointer {
  cursor: pointer;
}
