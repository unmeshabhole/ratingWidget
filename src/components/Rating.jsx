import React from "react";
import "./Rating.scss";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faCheckCircle } from "@fortawesome/free-solid-svg-icons";
import RatingIcon from "./RatingIcon.jsx";

const Rating = () => {
  const [rating, setRating] = React.useState(0);

  const onSaveRating = (index) => {
    setRating(index);
  };

  return (
    <div>
      <div>
        <h1>Rating Widget</h1>
      </div>
      <div className="box flex">
        {[1, 2, 3, 4, 5].map((index) => {
          return (
            <RatingIcon
              key={index}
              index={index}
              rating={rating}
              onSaveRating={onSaveRating}
            />
          );
        })}
      </div>
      <div id="thanksDiv" className={rating > 0 ? "showDiv" : "hideDiv"}>
        <FontAwesomeIcon icon={faCheckCircle} style={{ color: "green" }} />
        <span>Thanks! You rated this {rating} stars.</span>
      </div>
    </div>
  );
};
export default Rating;
