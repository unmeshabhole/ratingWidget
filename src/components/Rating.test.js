import React from "react";
import { shallow } from "enzyme";
import Rating from "./Rating";
import RatingIcon from "./RatingIcon";

test("Rating component html matches the snapshot ", () => {
  const wrapper = shallow(<Rating />);
  expect(wrapper.getElements()).toMatchSnapshot();
});

test("Rating component should render 5 Rating Icons", () => {
  const wrapper = shallow(<Rating />);
  expect(wrapper.find(RatingIcon)).toHaveLength(5);
});

test("Rating component should render a div with classname hideDiv at the start", () => {
  const wrapper = shallow(<Rating />);
  expect(wrapper.find(".hideDiv")).toHaveLength(1);
});
