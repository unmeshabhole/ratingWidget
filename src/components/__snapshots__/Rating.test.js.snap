// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rating component html matches the snapshot  1`] = `
Array [
  <div>
    <div>
      <h1>
        Rating Widget
      </h1>
    </div>
    <div
      className="box flex"
    >
      <RatingIcon
        index={1}
        onSaveRating={[Function]}
        rating={0}
      />
      <RatingIcon
        index={2}
        onSaveRating={[Function]}
        rating={0}
      />
      <RatingIcon
        index={3}
        onSaveRating={[Function]}
        rating={0}
      />
      <RatingIcon
        index={4}
        onSaveRating={[Function]}
        rating={0}
      />
      <RatingIcon
        index={5}
        onSaveRating={[Function]}
        rating={0}
      />
    </div>
    <div
      className="hideDiv"
      id="thanksDiv"
    >
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon={
          Object {
            "icon": Array [
              512,
              512,
              Array [],
              "f058",
              "M504 256c0 136.967-111.033 248-248 248S8 392.967 8 256 119.033 8 256 8s248 111.033 248 248zM227.314 387.314l184-184c6.248-6.248 6.248-16.379 0-22.627l-22.627-22.627c-6.248-6.249-16.379-6.249-22.628 0L216 308.118l-70.059-70.059c-6.248-6.248-16.379-6.248-22.628 0l-22.627 22.627c-6.248 6.248-6.248 16.379 0 22.627l104 104c6.249 6.249 16.379 6.249 22.628.001z",
            ],
            "iconName": "check-circle",
            "prefix": "fas",
          }
        }
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        style={
          Object {
            "color": "green",
          }
        }
        swapOpacity={false}
        symbol={false}
        title=""
        transform={null}
      />
      <span>
        Thanks! You rated this 
        0
         stars.
      </span>
    </div>
  </div>,
]
`;
